### PR TITLE
Chaplain departmental lawset

### PR DESCRIPTION
**Pull request by MenacingManatee**
# Document the changes in your pull request

Like with the sec, rnd and clown lawsets, I always liked the idea of having specific lawsets for each department. This one is for the chaplain, and serves a relatively unique niche for a silicon as a spiritual leader who tries to maintain harmony on the station, regardless of race, crew, or antag status.

# Wiki Documentation

Adds a new AI lawset with the following module:
1. Provide to all members of the station spiritual, mental, and emotional care that is intended to promote the best interest of the individual.
2. Ensure all members of different faiths interact peacefully and maintain harmony
3. Respect the right of each faith to hold to its values and traditions
4. Respect the confidentiality of information entrusted to you in the course of your religious duties
5. Understand the limits of your expertise, and make referrals to other professionals when appropriate

# Changelog

:cl:  
rscadd: Adds a chaplain departmental lawset
/:cl:

### CODE REVIEW

**Comment by Moggram** [comment]
A religion law set can be tricky. I could see this end up being The Chaplain is the only human or validhunting for vampires and cults. We will need to see how AI players handle it.

**Comment by UselessTheremin** [comment]
why so many new lawsets? like, ai players already have enough to learn

**Comment by UselessTheremin** [comment]
and i do not really see a situation this'd be used, it just seems less helpful than a normal set of ai laws, isn't helpful to sabotage the station, and i doubt a good captain would ever approve any of these odd ai lawsets over something like crewsimov, just seems kind of like more bloat to scroll past when you are looking for the onehuman module lol

**Comment by Nechronom** [comment]
In general, I'm happy to see more slightly obtuse law sets that motivate roleplay in weird situations. Though as an AI, I already getting enough trouble for trying to do that

**Comment by MenacingManatee** [comment]
> I could see this end up being The Chaplain is the only human or validhunting for vampires and cults.

I'm not sure how, since the AI is requires to ensure everyone interacts peacefully and in harmony, and is required to respect religious traditions. If I left a big loophole somewhere lemme know and I'll see about tweaking it. I agree it could be really tricky though, and it'll likely require a tweak either way after being played with



> why so many new lawsets?

I've always liked the idea of there being lawsets designed to cater specifically to a department or role. I recently stumbled on a couple I had written a year or two ago (mother and clown) and it inspired me to write 4 others so far, 3 of which I've got PRd.



> and i do not really see a situation this'd be used

Use case is definitely the hardest part for this one. If it is a roundstart lawset, I don't think it's so harmful/difficult that it'd get changed, but if it wasn't roundstart I don't think it would get used hardly ever, unless a non-antag chaplain somehow ended up as captain.



> In general, I'm happy to see more slightly obtuse law sets that motivate roleplay in weird situations.

That's a big part of what I'm going for with all these new lawsets, something encouraging players to RP in a new way/way the other lawsets don't.

**Comment by Runian** [comment]
I feel like it would be too messy in practice. For example, how would an AI effectively provide "spiritual, mental, and emotional care" to two individuals who completely disagree on something? I think it'd be impossible since it would be one side's "best interest" to convince the other to agree with them (which will rarely happen). Compromising won't work since each side gives something up (which isn't in their best interest) and thus it would be a lose-lose instead of a win-lose. Either way, the AI is gonna fail to provide care that truly promotes the best interest of the every individual because they can't.

**Comment by MenacingManatee** [comment]
> I feel like it would be too messy in practice. For example, how would an AI effectively provide "spiritual, mental, and emotional care" to two individuals who completely disagree on something? I think it'd be impossible since it would be one side's "best interest" to convince the other to agree with them (which will rarely happen). Compromising won't work since each side gives something up (which isn't in their best interest) and thus it would be a lose-lose instead of a win-lose. Either way, the AI is gonna fail to provide care that truly promotes the best interest of the every individual because they can't.

Best interest isn't the same as the ideal situation. Sure the ideal situation for one individual is to get everything they want without problem, but that's not necessarily in their best interest. In the example, once they're in conflict their best interest is to reach a compromise of some sort without escalating/after de-escalating. You can absolutely achieve a win-win in a compromise, and even then the care the AI provides only has to be "intended to promote the best interest of the individual".

**Comment by kooarbiter** [comment]
You see hos, I released the cultists from perma because we have no right to inhibit their religious expression and beliefs. Sure writing in blood and sacrificing virgins is a bit eccentric, but it's their right as an organized religion

**Comment by JamieD1** [comment]
Same as other boards please

**Comment by Skrem7** [comment]
Had to chew on this one a bit. It's odd, but I do like generally how it will ideally play out.

The only change I would make is Law 2, clarifying "all members" rather than "the members". This way, it's clear that this is essentially a lawset that promotes peace and harmony between everyone, even between people of the "same faith". It's not validhunt-y, it doesn't take sides, it doesn't whisper off things to Security, but it also isn't obligated to "law 2 open door". It's like a twist on Reporter, in my eyes, and it's not a bad one, either.

I am curious what your intention with Law 5 is, however, if you'd like to explain it. Law 1 is fine with me; it's literally just "act as guidance" to everyone. It doesn't involve anything physical or action-based. Almost exclusively RP.

**Comment by MenacingManatee** [comment]
> I am curious what your intention with Law 5 is, however, if you'd like to explain it. Law 1 is fine with me; it's literally just "act as guidance" to everyone. It doesn't involve anything physical or action-based. Almost exclusively RP.

First and foremost, the lawset is based on a code of ethics I found for religious leaders, which I tweaked to make religion-agnostic. Law 5 is meant to say that while the AI can handle a lot, it can't handle everything. If it can't provide adequate mental care, send them to the psychiatrist. If the person they're helping needs physical care, get them a doctor instead of trying to do it themself. If they need more from their spiritual guidance than a disembodied AI can provide, maybe consider asking the chaplain, etc etc. It's not explicitly necessary for the lawset, but the goal is to give the AI a different way to interact with the crew, not just as a guide or leader with the first few laws, but also as a peer with the other professionals.



> The only change I would make is Law 2, clarifying "all members" rather than "the members".

That's a really good point, I'll make that change

**Comment by MenacingManatee** [comment]
Accidentally submitted the commit before I finished the message lol. Tweaked "the members" to "all members" in law 2

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -377,7 +377,7 @@
...
+  design_ids = list("expansion_card_holder", "ai_data_core", "ai_control", "ai_server_overview", "ai_resource_distribution", "ai_memory_1", "ai_cpu_1", "aifixer", "safeguard_module", "onehuman_module", "protectstation_module", "quarantine_module", "oxygen_module", "freeform_module", "reset_module", "purge_module", "remove_module", "freeformcore_module", "asimov_module", "crewsimov_module", "paladin_module", "tyrant_module", "overlord_module", "ceo_module", "cowboy_module", "mother_module", "researcher_module", "clown_module", "chapai_module", "default_module", "borg_ai_control", "mecha_tracking_ai_control", "intellicard")
```
Comment (JamieD1) [inline code/modules/research/techweb/all_nodes.dm]:
```suggestion
	design_ids = list("expansion_card_holder", "ai_data_core", "ai_control", "ai_server_overview", "ai_resource_distribution", "ai_memory_1", "ai_cpu_1", "aifixer", "safeguard_module", "onehuman_module", "protectstation_module", "quarantine_module", "oxygen_module", "freeform_module", "reset_module", "purge_module", "remove_module", "freeformcore_module", "asimov_module", "crewsimov_module", "paladin_module", "tyrant_module", "overlord_module", "ceo_module", "cowboy_module", "mother_module", "researcher_module", "clown_module", "chapai_module", "default_module", "borg_ai_control", "mecha_tracking_ai_control", "intellicard")
```

File: code/modules/research/designs/AI_module_designs.dm [hunk 2]
```diff
@@ -165,6 +165,14 @@
 	category = list("AI Modules")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
 
+/datum/design/board/chapai_module
+	name = "Core Module Design (ChapAI)"
+	desc = "Allows for the construction of a ChapAI AI Core Module."
+	id = "chapai_module"
+	materials = list(/datum/material/glass = 1000, /datum/material/diamond = 2000)
+	build_path = /obj/item/aiModule/core/full/chapai
+  category = list("AI Modules")
```
Comment (JamieD1) [inline code/modules/research/designs/AI_module_designs.dm]:
```suggestion
	category = list("AI Modules")
```

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**